docs/source/tutorial/basic.rst

```diff
@@ -314,24 +326,43 @@ In this case, :meth:`~Link.zerograds` is automatically called by the update meth
    ...     return loss
    >>> optimizer.update(lossfun, args...)
 
-Some parameter/gradient manipulations, e.g. weight decay and gradient clipping, can be done by setting *hook functions* to the optimizer.
-Hook functions are called by the :meth:`~Optimizer.update` method in advance of the actual update.
-For example, we can set weight decay regularization by running the next line beforehand:
 
-.. doctest::
+Trainer
+~~~~~~~
 
-   >>> optimizer.add_hook(chainer.optimizer.WeightDecay(0.0005))
+When we want to train neural networks, we have to run *training loops* that update parameters for many times.
```

Comment:
It seems for is not needed in this case (c.f. [here](http://ell.stackexchange.com/questions/11362/many-times-or-for-many-times))